plugins/modules/nsupdate.py

```diff
@@ -370,7 +370,8 @@ def modify_record(self):
             except (socket_error, dns.exception.Timeout) as e:
                 self.module.fail_json(msg='DNS server error: (%s): %s' % (e.__class__.__name__, to_native(e)))
 
-            entries_to_remove = [n.to_text() for n in lookup.answer[0].items if n.to_text() not in self.value]
+            lookup_result = lookup.answer[0] if lookup.answer else lookup.authority[0]
+            entries_to_remove = [n.to_text() for n in lookup_result.items if n.to_text() not in self.value]
```

Comment:
Hmm, in case `lookup.answer` does not exist, shouldn't `entries_to_remove` simply be set to `[]`? In that situation there is no current `NS` value on that level that can be deleted, after all.

So something like
```suggestion
            entries_to_remove = [n.to_text() for n in lookup.answer[0].items if n.to_text() not in self.value] if lookup.answer else []
```
maybe?

Reply:
my understanding of that block is bind9 will not let you rewrite the last NS record in the zone, so the new record is inserted first, and old one is deleted after. In my case it would not be the last, but the record does exist and needs to be deleted. But since the zone is subdelegated to another server, the data is in authority section. My guess is it was tested on the server where delegation was to another local zone so this was missed. If we do it like you suggest, we would have dupes.

Reply:
Hmm, ok. This still doens't sound right to me, but I don't use bind9 or the module so I don't know what really happens :)

In any case, this shouldn't break any existing behavior since if this happened so far, it resulted in an error, so in the worst case this is introducing wrong behavior in a case that didn't work at all before. If someone complains about it, we'll have to look into it again to figure out why exactly it works for you but not for them. And if nobody complains, I guess it's fine for everyone then :)